src/mimeparser.rs

```diff
@@ -376,6 +376,12 @@ impl MimeMessage {
         if !encrypted {
             signatures.clear();
         }
+        if !signatures.is_empty() {
+            if let Some(peerstate) = &mut decryption_info.peerstate {
+                peerstate.prefer_encrypt = EncryptPreference::Mutual;
+                peerstate.save_to_db(&context.sql).await?;
```

Comment:
These two lines should prob. be wrapped in `if peerstate.prefer_encrypt != EncryptPreference::Mutual` since `save_to_db` is a database access that will otherwise happen everytime an encrypted message arrives.